### PR TITLE
Add channel changer script

### PR DESCRIPTION
**Pull request by speakerdamage**
new script using Glut

### CODE REVIEW

**Comment by markwheeler** [comment]
I think you may want to limit how often you call `redraw()` as the encoder calls can come in pretty fast.

A solution I've used is to have a metro running at the screen framerate and set a dirty flag:
```lua
local screen_dirty = true
```
In `init()` you could have:
```lua
local SCREEN_FRAMERATE = 15
local screen_refresh_metro = metro.alloc()
screen_refresh_metro.callback = function()
  if screen_dirty then
    screen_dirty = false
    redraw()
  end
end
screen_refresh_metro:start(1 / SCREEN_FRAMERATE)
```
And then just set `screen_dirty = true` everywhere you currently call `redraw()`

If there's still performance issues drawing at 15fps then you could optimize the `redraw()` function quite a bit - right now it's generating 8192 random numbers every frame but I think you could get a similar effect by, for example, generating a table of noise once and then just randomly offsetting or shuffling the rows.

**Comment by speakerdamage** [comment]
thanks for this. my plan was/is to have a much simpler animation on encoder ```redraw()``` but if I don't do that I'll add that dirty flag. I'll also work on optimizing the random screen.

File: scripts/speakerdamage/channel_changer.lua [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+-- ENC 1: volume
+-- ENC 2: speed
+-- ENC 3: pitch
+--
+-- your display is not broken
+
+engine.name = 'Glut'
+local VOICES = 1
+voiceGate = 0
+shiftMode = 0
```
Comment (pq) [inline scripts/speakerdamage/channel_changer.lua]: can this be local or is it getting read elsewhere?

File: scripts/speakerdamage/channel_changer.lua [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+-- KEY 1 (HOLD): tv guide
+-- ENC 1: volume
+-- ENC 2: speed
+-- ENC 3: pitch
+--
+-- your display is not broken
+
+engine.name = 'Glut'
+local VOICES = 1
+voiceGate = 0
```
[the remark this answers is not in the document]
Comment (speakerdamage) [reply scripts/speakerdamage/channel_changer.lua]: nope, will remove.

File: scripts/speakerdamage/channel_changer.lua (hunk 1 above)
Comment (speakerdamage) [reply scripts/speakerdamage/channel_changer.lua]: I think local should be fine

File: scripts/speakerdamage/channel_changer.lua [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+shiftMode = 0
+grainPosition = 0
```
[the remark this answers is not in the document]
Comment (speakerdamage) [reply scripts/speakerdamage/channel_changer.lua]: I think I can lose this - will check when I'm in front of norns again.

**Comment by markwheeler** [comment]
Thanks for looking at these bits. It looks like in your latest revision you added in the screen_refresh_metro code but you're still calling `redraw()` - for this to work you need to replace all of those calls with setting `screen_dirty` to true. However, I also see that you removed redrawing from the encoder callbacks completely which is really what this suggestion was meant to solve (the UI freezing when turning an encoder quickly).

**Comment by speakerdamage** [comment]
I should have waited to push those changes together since I didn't have time to really look at redraw() - my bad. that solution has been removed for now. hopefully will have some time this week to take another look.

**Pull request merged**